### PR TITLE
[fix]:[CDS-81220]:Moved V2 logic from S3BucketResource to ArtifactResourceUtils

### PR DESCRIPTION
**Pull request by Rakshitharness**
## Describe your changes
Moved some V2 logic of S3bucket resource to ArtifactResourceUtils and added the Unit test for the same.
Done the basic testing for the S3 
- Check all the drop down in v1 and V2
- Run the pipeline while given all the values
- run the pipeline while giving it as a expression <+lastPublished.tag>.regex()

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55079)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/buckets/resources/s3/S3BucketResource.java [hunk 1]
```diff
@@ -184,42 +181,9 @@ public ResponseDTO<List<BucketResponseDTO>> getBucketsV2WithServiceV2(@QueryPara
       @QueryParam(NGCommonEntityConstants.SERVICE_KEY) String serviceRef,
       @BeanParam GitEntityFindInfoDTO gitEntityBasicInfo) {
     // In case of ServiceV2 Calls
-    if (StringUtils.isNotBlank(serviceRef)) {
-      final ArtifactConfig artifactSpecFromService = artifactResourceUtils.locateArtifactInService(
-          accountId, orgIdentifier, projectIdentifier, serviceRef, fqnPath);
-
-      AmazonS3ArtifactConfig amazonS3ArtifactConfig = (AmazonS3ArtifactConfig) artifactSpecFromService;
-
```
Comment (Abhishek-CDC) [inline 120-ng-manager/src/main/java/io/harness/ng/core/buckets/resources/s3/S3BucketResource.java:192]: Other methods also having logic in resource file.
Lets move them as well

**Comment by Rakshitharness** [comment]
trigger uts

**Comment by Rakshitharness** [comment]
trigger ut9

File: 120-ng-manager/src/main/java/io/harness/ng/core/buckets/resources/s3/S3BucketResource.java [hunk 2]
```diff
@@ -274,62 +192,9 @@ public ResponseDTO<List<FilePathDTO>> getFilePathsForServiceV2(@QueryParam("regi
...
-            .getValue();
-
-    // Common logic in case of ServiceV1 and ServiceV2
-    IdentifierRef connectorRef = IdentifierRefHelper.getIdentifierRef(
-        resolvedAwsConnectorIdentifier, accountId, orgIdentifier, projectIdentifier);
-
-    List<BuildDetails> s3ArtifactPaths = s3ResourceService.getFilePaths(
-        connectorRef, resolvedRegion, resolvedBucketName, fileFilter, orgIdentifier, projectIdentifier);
+    List<BuildDetails> s3ArtifactPaths = artifactResourceUtils.getFilePathsForServiceV2S3(region,
+        awsConnectorIdentifier, bucketName, filePathRegex, fileFilter, accountId, orgIdentifier, projectIdentifier,
```
Comment (naman-goenka) [inline 120-ng-manager/src/main/java/io/harness/ng/core/buckets/resources/s3/S3BucketResource.java:196]: nit: we can abstract scope identifiers to scope object to reduce method params

**Review by naman-goenka — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/core/buckets/resources/s3/S3BucketResource.java (hunk 1 above)
Comment (Abhishek-CDC) [reply 120-ng-manager/src/main/java/io/harness/ng/core/buckets/resources/s3/S3BucketResource.java:192]: getBucketsV2
v2/getBuckets
getFilePaths
v2/getFilePaths
Above endPoint methods having logic in resource file

**Comment by Rakshitharness** [comment]
trigger uts

**Review by Abhishek-CDC — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Comment by Rakshitharness** [comment]
trigger review

**Pull request merged**